### PR TITLE
Allow $ and @ characters in field names.

### PR DESCRIPTION
**Pull request by fmassot**
Motivated by
- user who needs to migrate data from MongoDB into Quickwit ($ is often used in mongo I believe)
- elasticsearch often used `@`  as the first character of field names, I got the issue a few months ago for the `@timestamp` field

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mod.rs [hunk 1]
```diff
@@ -146,6 +145,7 @@ mod tests {
         assert!(validate_field_mapping_name("my-field").is_ok());
         assert!(validate_field_mapping_name("my.field").is_ok());
         assert!(validate_field_mapping_name("my_field").is_ok());
+        assert!(validate_field_mapping_name("my_field").is_ok());
```
Comment (etolbakov) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/mod.rs:148]: are L147 and L148 the same?

Comment (fmassot) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/mod.rs:148]: yes :) thanks!

File: docs/configuration/index-config.md [hunk 2]
```diff
@@ -491,12 +491,13 @@ src.port:53 AND query_params.ctk:e42bb897d
...
 
 Currently Quickwit only accepts field name that matches the following regular expression:
-`[a-zA-Z][_\.\-a-zA-Z0-9]*$`
+`^[@$_\-a-zA-Z][@$_\.\-a-zA-Z0-9]{0,254}$`
 
 In plain language:
 - it needs to have at least one character.
-- it should only contain latin letter `[a-zA-Z]` digits `[0-9]` or (`.`, `-`, `_`).
-- the first character needs to be a letter.
+- it can only contain uppercase and lowercase ASCII letters `[a-zA-Z]`, digits `[0-9]`, hyphens `-`, `$`, `@` and underscores `_`;
```
Comment (guilload) [inline docs/configuration/index-config.md:498]: You removed the `.`

Comment (fmassot) [reply docs/configuration/index-config.md:498]: :/

File: docs/configuration/storage-config.md [hunk 3]
```diff
@@ -146,3 +146,5 @@ storage:
     flavor: minio
     endpoint: http://127.0.0.1:9000
 ```
+
+Note: `default_index_root_uri` or index URIs do not include the endpoint, you should set it as a typical S3 path such as `s3://indexes`.
```
Comment (fmassot) [inline docs/configuration/storage-config.md:150]: Just a minor addition in the doc as I saw 2 users making this mistake, that is setting the URI as `s3://127.0.0.1:9000/indexes`

**Review by guilload — APPROVED** [review]

**Pull request merged**